### PR TITLE
Add support for schemas of schemas

### PR DESCRIPTION
**Pull request by bugagashenkj**

### CODE REVIEW

File: applications/invalid-schemas/config/example.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  parameter: 'value',
+
+  subSection: {
+    parameter: 'value'
+  }
+
+}
```
Comment (lundibundi) [inline applications/invalid-schemas/config/example.js]: I think this one can be removed.

File: applications/invalid-schemas/schemas/CategoryName.category [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  Name: { domain: 'Nomen', display: true, required: true, unique: true },
+  CustomNomen: { domain: 'CustomNomen', display: true, required: true },
+}
+
```
Comment (lundibundi) [inline applications/invalid-schemas/schemas/CategoryName.category]: Unnecessary newline?

File: applications/invalid-schemas/schemas/config.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+async () => (
+  !api.test.useSystemSchemas ?
```
Comment (lundibundi) [inline applications/invalid-schemas/schemas/config.js]: Please add a comment explaining this ternary.

File: applications/invalid-schemas/test/schemas-validation.test.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+const { ValidationError, MetaschemaError } = api.metaschema.errors;
+
+api.test.useSystemSchemas = true;
+application.loadSchemas(error => {
+  test.endAfterSubtests();
+  test.test('schemas validation', test => {
```
Comment (lundibundi) [inline applications/invalid-schemas/test/schemas-validation.test.js]: Why additional subtest and not just (top-level) `test.strictSame`?

File: applications/invalid-schemas/schemas/config.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+async () => (
+  // load schemas of schemas after start application to catch the error
+  !api.test.useSystemSchemas ?
+    api.globalstorage.schemaConfig :
+    {
```
Comment (nechaido) [inline applications/invalid-schemas/schemas/config.js]: Why do you destructure and then collect the same object without any changes?

File: applications/invalid-schemas/schemas/config.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+      ...await api.metaschema.fs.applySystemConfig(
+        api.path.resolve(
```
Comment (nechaido) [inline applications/invalid-schemas/schemas/config.js]:
```suggestion
        api.path.join(application.dir, 'validationSchemas')
```

**Comment by tshemsedinov** [comment]
ping @belochub

**Comment by tshemsedinov** [comment]
@bugagashenkj please rebase on master

**Comment by bugagashenkj** [comment]
@tshemsedinov it can be landed.

**Comment by bugagashenkj** [comment]
@tshemsedinov

**Comment by tshemsedinov** [comment]
Not relevant for v2.x

**Pull request closed without merging**